x/tokenomics/keeper/settle_pending_claims.go

```diff
@@ -32,45 +32,49 @@ func (k Keeper) SettlePendingClaims(ctx cosmostypes.Context) (
 	logger := k.Logger().With("method", "SettlePendingClaims")
 	logger.Debug("settling expiring claims")
 
-	// Retrieve all expiring claims as an iterator.
-	// DEV_NOTE: This previously retrieve a list but has been change to account for large claim counts.
-	expiringClaimsIterator := k.GetExpiringClaimsIterator(ctx)
-	defer expiringClaimsIterator.Close()
-
-	blockHeight := ctx.BlockHeight()
-
 	// Initialize results structs.
+	blockHeight := ctx.BlockHeight()
 	settledResults = make(tlm.ClaimSettlementResults, 0)
 	expiredResults = make(tlm.ClaimSettlementResults, 0)
 	settlementContext := NewSettlementContext(ctx, &k, logger)
 	numExpiringClaims := 0
 	numDiscardedFaultyClaims = 0
 
+	// Retrieve an iterator of all expiring claims.
+	// DEV_NOTE: This previously retrieve a list but has been change to account for large claim counts.
+	expiringClaimsIterator := k.GetExpiringClaimsIterator(ctx, settlementContext, blockHeight)
+	defer expiringClaimsIterator.Close()
+
 	// Iterating over all potentially expiring claims.
 	// This loop does the following:
-	// - Retrieve the claim
-	// - Settle the claim
-	// - Remove the claim from the state
-	// - Emit an event
-	// - Update the relevant actors in the state
+	// 1. Retrieve the claim
+	// 2. Settle the claim
+	// 3. Remove the claim from the state
+	// 4. Emit an event
+	// 5. Update the relevant actors in the state
 	for ; expiringClaimsIterator.Valid(); expiringClaimsIterator.Next() {
 		claim, iterErr := expiringClaimsIterator.Value()
 		if iterErr != nil {
-			claimKey := string(expiringClaimsIterator.Key())
-			logger.Error(tokenomicstypes.ErrTokenomicsSettlementInternal.Wrapf(
-				"[UNEXPECTED ERROR] Critical error during claim settlement during session with key %s. Claim will be discarded to prevent chain halt: %s",
-				claimKey, err,
-			).Error())
 			numDiscardedFaultyClaims++
+			claimKey := string(expiringClaimsIterator.Key())
+			claimErr := tokenomicstypes.ErrTokenomicsSettlementInternal.Wrapf(
+				"[UNEXPECTED ERROR] Critical error during claim settlement during session with key %s. Claim will be discarded to prevent chain halt: %v",
```

Comment:
Prefer `%s` over `%v` when interpolating strings to avoid error type paths from appearing in error logs.
```suggestion
				"[UNEXPECTED ERROR] Critical error during claim settlement during session with key %s. Claim will be discarded to prevent chain halt: %s",
```

Reply:
This is demonstrable with the following test:

<details>
<summary>Go test...</summary>

```go
package poktroll_test

import (
	"fmt"
	"testing"

	sdkerrors "cosmossdk.io/errors"
)

var myErr = sdkerrors.Register("codespace", 42069, "description")

// TestErrorInterpolationBehavior demonstrates how %v interpolation can leak
// variable path information in error messages, but only at the second level.
func TestErrorInterpolationBehavior(t *testing.T) {
	t.Log("=== Error Interpolation Demonstration ===")
	t.Log("This demonstrates how v interpolation can leak variable paths in error messages.")

	// Create a base error
	baseErr := fmt.Errorf("base error message")
	t.Logf("1. base error: %s", baseErr)

	// First level wrapping - comparing %s vs %v
	stringErr := myErr.Wrapf("string interpolation: %s", baseErr)
	varErr := myErr.Wrapf("var interpolation: %v", baseErr)

	t.Logf("2. fist level %%s: %s", stringErr)
	t.Logf("3. first level %%v: %s", varErr)

	// Second level wrapping - this is where the issue appears
	secondLevelVarVar := fmt.Errorf("second level %%v then %%v: %v", varErr)
	secondLevelVarString := fmt.Errorf("second level %%v then %%s: %s", varErr)
	secondLevelStringVar := fmt.Errorf("second level %%s then %%v: %v", stringErr)
	secondLevelStringString := fmt.Errorf("second level %%s then %%s: %s", stringErr)

	t.Logf("4. %s", secondLevelVarVar)
	t.Logf("5. %s", secondLevelVarString)
	t.Logf("6. %s", secondLevelStringVar)
	t.Logf("7. %s", secondLevelStringString)
}

```

</details>

<img width="1200" height="286" alt="image" src="https://github.com/user-attachments/assets/e8a90eb9-e33d-4f43-aba5-4f1eb7c77ae2" />